From 7bef564d8cab4e90ae61a7400bd5a454690ddc2f Mon Sep 17 00:00:00 2001
From: Katherine Lu <katherine.lu@freescale.com>
Date: Tue, 20 Jan 2009 15:50:27 +0800
Subject: [PATCH] ENGR00105614 MX35: update switch configure for bluetooth pcm and spdif for TO2

Update switch configure for bluetooth pcm and spdif for MX35 TO2

Signed-off-by: Katherine Lu <katherine.lu@freescale.com>
---
 arch/arm/mach-mx35/mx35_3stack_gpio.c |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx35/mx35_3stack_gpio.c b/arch/arm/mach-mx35/mx35_3stack_gpio.c
index 0c1cc46..957e99f 100644
--- a/arch/arm/mach-mx35/mx35_3stack_gpio.c
+++ b/arch/arm/mach-mx35/mx35_3stack_gpio.c
@@ -872,7 +872,10 @@ void gpio_spdif_active(void)
 			  | PAD_CTL_100K_PU | PAD_CTL_HYS_SCHMITZ);
 	/* SPDIF ext clock */
 	mxc_request_iomux(MX35_PIN_SCK5, MUX_CONFIG_ALT1);
-	pmic_gpio_set_bit_val(MCU_GPIO_REG_RESET_2, 0, 1);
+	if (board_is_mx35(BOARD_REV_2))
+		pmic_gpio_set_bit_val(MCU_GPIO_REG_GPIO_CONTROL_2, 5, 1);
+	else
+		pmic_gpio_set_bit_val(MCU_GPIO_REG_RESET_2, 0, 1);
 }
 
 EXPORT_SYMBOL(gpio_spdif_active);
@@ -934,7 +937,10 @@ void gpio_activate_bt_audio_port(void)
 	mxc_iomux_set_pad(MX35_PIN_SRXD5, pad_val);
 	mxc_iomux_set_pad(MX35_PIN_SCK5, pad_val);
 	mxc_iomux_set_pad(MX35_PIN_STXFS5, pad_val);
-	pmic_gpio_set_bit_val(MCU_GPIO_REG_RESET_2, 0, 0);
+	if (board_is_mx35(BOARD_REV_2))
+		pmic_gpio_set_bit_val(MCU_GPIO_REG_GPIO_CONTROL_2, 5, 0);
+	else
+		pmic_gpio_set_bit_val(MCU_GPIO_REG_RESET_2, 0, 0);
 }
 
 EXPORT_SYMBOL(gpio_activate_bt_audio_port);
-- 
1.5.4.4

